scilpy/tractograms/intersection_finder.py

```diff
@@ -0,0 +1,220 @@
+        collided = np.full((len(streamlines)), False, dtype=np.bool_)
+        excluded = np.full((len(streamlines)), False, dtype=np.bool_)
+
+        # si   : Streamline Index | index of streamline within the tractogram.
+        # pi   : Point Index      | index of point coordinate within a
+        #                           streamline.
+        # segi : Segment Index    | index of streamline segment within the
+        #                           entire tractogram.
+        for segi, center in v_enumerate(self.seg_centers,
+                                        self.verbose):
+            si = self.seg_indices[segi][0]
+
+            # [Pruning 1] If current streamline has already collided or been
+            #             excluded, skip.
+            if colliders[si] or excluded[si]:
+                continue
+
+            neighbors = self.tree.query_ball_point(center,
+                                                   self.max_seg_length,
+                                                   workers=-1)
+
+            for neighbor_segi in neighbors:
+                neighbor_si = self.seg_indices[neighbor_segi][0]
+
+                # [Pruning 2] Skip if neighbor is our streamline
+                if neighbor_si == si:
+                    continue
+
+                # [Pruning 3] If neighbor has already collided or been
+                #             excluded, skip.
+                if colliders[neighbor_si] or excluded[neighbor_si]:
+                    continue
+
+                p0 = streamlines[si][self.seg_indices[segi][1]]
+                p1 = streamlines[si][self.seg_indices[segi][1] + 1]
+                q0 = streamlines[neighbor_si][
+                    self.seg_indices[neighbor_segi][1]]
+                q1 = streamlines[neighbor_si][
+                    self.seg_indices[neighbor_segi][1] + 1]
+
+                rp = self.diameters[si] / 2
+                rq = self.diameters[neighbor_si] / 2
+
+                distance, _, p_coll, _ = dist_segment_segment(p0, p1, q0, q1)
+                collide = distance <= rp + rq + self.FLOAT_EPSILON
+                external_distance = distance - rp - rq
+
+                if collide:
+                    colliders[si] = True
+                    collisions[si] = p_coll
+                    collided[neighbor_si] = True
+                    break
+
+                if (tmv_size_threshold is not None and
+                        external_distance < (tmv_size_threshold /
+                                             (math.sqrt(2) / 2))):
+                    excluded[si] = True
+                    break
+
+        logging.debug("Finished finding intersections in " +
+                      str(round(time.time() - start_time, 2)) + " seconds.")
+
+        self._colliders = colliders
+        self._collisions = collisions
+        self._collided = collided
+        self._excluded = excluded
+
+    def build_tractograms(self, args):
```

Comment:
you don't need the whole `args` namespace. You only use `save_colliders` and `save_collided`.